cpp/tests/io/json_test.cpp

```diff
@@ -1563,4 +1606,23 @@ TEST_P(JsonReaderParamTest, JsonDtypeParsing)
   }
 }
 
+TYPED_TEST(JsonFixedPointReaderTest, SingleColumnNegativeScale)
+{
+  this->run_tests({"1.23", "876e-2", "5.43e1", "-0.12", "0.25", "-0.23", "-0.27", "0.00", "0.00"},
```

Comment:
Shouldn't we add a test for empty strings?

Reply:
Having an empty value in these tests does not work well because the reference `to_fixed_point` converts them to zero, old JSON reader reads them as null, and the new reader throws.
Added a separate test for this case.